tests/collections/asr/decoding/test_rnnt_decoding.py

```diff
@@ -0,0 +1,252 @@
+
+            #     print(f"Tokens at timestep {t} = {t_u}")
+            # print()
+
+    @pytest.mark.skipif(
+        not NUMBA_RNNT_LOSS_AVAILABLE, reason='RNNTLoss has not been compiled with appropriate numba version.',
+    )
+    @pytest.mark.with_downloads
+    @pytest.mark.unit
+    @pytest.mark.parametrize(
+        "beam_config",
+        [
+            {"search_type": "greedy"},
+            {"search_type": "default", "beam_size": 2,},
+            {"search_type": "alsd", "alsd_max_target_len": 0.5, "beam_size": 2,},
+            {"search_type": "tsd", "tsd_max_sym_exp_per_step": 3, "beam_size": 2,},
+            {"search_type": "maes", "maes_num_steps": 2, "maes_expansion_beta": 2, "beam_size": 2},
+            {"search_type": "maes", "maes_num_steps": 3, "maes_expansion_beta": 1, "beam_size": 2},
+        ],
+    )
+    def test_beam_decoding_preserve_alignments(self, test_data_dir, beam_config):
+        beam_size = beam_config.pop("beam_size", 1)
+        model, encoded, encoded_len = get_model_encoder_output(test_data_dir, 'stt_en_conformer_transducer_small')
+        beam = beam_decode.BeamRNNTInfer(
+            model.decoder,
+            model.joint,
+            beam_size=beam_size,
+            return_best_hypothesis=False,
+            preserve_alignments=True,
+            **beam_config,
+        )
+
+        enc_out = encoded
+        enc_len = encoded_len
+        blank_id = torch.tensor(model.joint.num_classes_with_blank - 1, dtype=torch.int32)
+
+        with torch.no_grad():
+            hyps = beam(encoder_output=enc_out, encoded_lengths=enc_len)[0]  # type: rnnt_utils.Hypothesis
+            hyp, all_hyps = decode_text_from_nbest_hypotheses(hyps, model.decoding)
+            hyp = hyp[0]  # best hypothesis
+            all_hyps = all_hyps[0]
+
+            assert hyp.alignments is not None
+
+            if beam_config['search_type'] == 'alsd':
+                assert len(all_hyps) <= int(beam_config['alsd_max_target_len'] * float(enc_len[0]))
+
+            # Use the following commented print statements to check
+            # the alignment of other algorithms compared to the default
+            for idx, hyp_ in enumerate(all_hyps):  # type: (int, rnnt_utils.Hypothesis)
+                # print("Hyp index", idx + 1, "text :", hyp_.text)
+
+                # Alignment length (T) must match audio length (T)
+                assert len(hyp_.alignments) == enc_len[0]
+
+                for t in range(len(hyp_.alignments)):
+                    t_u = []
+                    for u in range(len(hyp_.alignments[t])):
+                        logp, label = hyp_.alignments[t][u]
+                        assert torch.is_tensor(logp)
+                        assert torch.is_tensor(label)
+
+                        t_u.append(int(label))
+
+                    # Blank token must be the last token in the current
+                    if len(t_u) > 1:
+                        assert t_u[-1] == blank_id
```

Comment:
Can we also add a check that all other tokens are non-blank? That was breaking in the bug-fix PR I did for batched alignment calculation, so might be a helpful check here as well

Reply:
Nice catch, will add.